src/lib.rs

```diff
@@ -47,6 +49,36 @@ pub struct TransferMessage {
     #[serde(with = "hex::serde")]
     pub recipient: EthAddress,
     pub valid_till_block_height: Option<u64>,
+    #[serde(
+        serialize_with = "serialize_hex_option_eth_address",
+        deserialize_with = "deserialize_hex_option_eth_address"
+    )]
+    pub aurora_sender: Option<EthAddress>,
+}
+
+fn serialize_hex_option_eth_address<S>(
```

Comment:
I propose refactoring the `EthAdress` type to something like this:

```
#[derive(BorshDeserialize, BorshSerialize)]
pub struct EthAddress([u8; 20]);

impl<'de> Deserialize<'de> for EthAddress {
    fn deserialize<D>(deserializer: D) -> Result<Self, <D as serde::Deserializer<'de>>::Error>
    where
        D: serde::Deserializer<'de>,
    {
        let mut s = <String as Deserialize>::deserialize(deserializer)?;
        if s.starts_with("0x") {
            s = s[2..].to_string();
        }
        let result = Vec::from_hex(&s).map_err(|err| serde::de::Error::custom(err.to_string()))?;
        Ok(EthAddress(result.try_into().unwrap()))
    }
}

impl Serialize for EthAddress {
    fn serialize<S>(&self, serializer: S) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
    where
        S: Serializer,
    {
        serializer.serialize_str(&hex::encode(self.0))
    }
}
```

Reply:
In that case we should fix all dependencies from EthAddress in protocol and services. Currently it is convenient that it is the same type as [u8; 20]